### PR TITLE
feat(web-analytics): Add mini-version of the retention cohort analysis

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

No retention cohort analysis on web analytics, and the UI that does exist is hard to fit in a small space.

## Changes

I added some options to the retention table, to make it work better in a smaller space.
* Hide the line graph
* Reduce the font size and some margin sizes
* Hide the size count

iPhone SE
Screenshot 2023-11-14 at 15 02 27

iPad
Screenshot 2023-11-14 at 15 03 29

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/web-analytics/WebDashboard.tsx [hunk 1]
```diff
@@ -120,7 +123,11 @@ const TabsTileItem = ({ tile }: { tile: TabsTile }): JSX.Element => {
 
     return (
         <WebTabs
-            className={`col-span-1 row-span-1 md:col-span-${layout.colSpan ?? 6} md:row-span-${layout.rowSpan ?? 1}`}
+            className={clsx(
+                'col-span-1 row-span-1',
+                ` md:col-span-${layout.colSpan ?? 6} md:row-span-${layout.rowSpan ?? 1}`,
```
Comment (benjackwhite) [inline frontend/src/scenes/web-analytics/WebDashboard.tsx:128]: nit
```suggestion
                `md:col-span-${layout.colSpan ?? 6} md:row-span-${layout.rowSpan ?? 1}`,
```

File: frontend/src/scenes/web-analytics/WebDashboard.tsx [hunk 2]
```diff
@@ -146,7 +153,7 @@ const WebQuery = ({ query }: { query: QuerySchema }): JSX.Element => {
 
 export const WebAnalyticsDashboard = (): JSX.Element => {
     return (
-        <div className="w-full flex flex-col pt-2">
+        <div className="web-analytics-page w-full flex flex-col pt-2">
```
Comment (benjackwhite) [inline frontend/src/scenes/web-analytics/WebDashboard.tsx:156]: For non-utility classes we should be using the other style
```suggestion
        <div className="WebAnalyticsDashboard w-full flex flex-col pt-2">
```

File: frontend/src/scenes/web-analytics/webAnalytics.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.web-analytics-page {
```
Comment (benjackwhite) [inline frontend/src/scenes/web-analytics/webAnalytics.scss:1]:
```suggestion
.WebAnalyticsDashboard {
```

**Review by benjackwhite — COMMENTED** [review]
Couple of small things but overall looks good! 

Would be cool if we could get my big refactor in first... https://github.com/PostHog/posthog/pull/18570

File: frontend/src/scenes/web-analytics/WebDashboard.tsx (hunk 2 above)
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/WebDashboard.tsx:156]: Fair enough, I just copied some other code we saw :D

File: frontend/src/scenes/web-analytics/webAnalytics.scss [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.WebAnalyticsDashboard {
+    // TODO remove this when we no longer use the deprecated Card component
+    .ant-card-body {
```
Comment (benjackwhite) [inline frontend/src/scenes/web-analytics/webAnalytics.scss:3]: I think this can be removed now

File: frontend/src/queries/schema.ts [hunk 5]
```diff
@@ -391,6 +392,18 @@ export interface SavedInsightNode extends Node, InsightVizNodeViewProps, DataTab
...
 
+/** Chart specific rendering options.
+ * Use ChartRenderingMetadata for non-serializable values, e.g. onClick handlers
+ * @see ChartRenderingMetadata
+ * **/
+export interface VizSpecificOptions {
+    [InsightType.RETENTION]?: {
+        hideLineGraph?: boolean
+        hideSizeColumn?: boolean
+        useSmallLayout?: boolean
```
Comment (benjackwhite) [inline frontend/src/queries/schema.ts:403]: This is the only property I don't think makes sense. 
We should do this with container queries given that it looks nicer regardless of the setting, it's more based on the container size anyways which applies no matter where it is rendered right?

**Review by benjackwhite — CHANGES_REQUESTED** [review]
Looks good. The one exception is the "small layout" thing. I think this is the kind of property that over time becomes hard to manage and rather the component itself should just do this with container queries

File: frontend/src/queries/schema.ts (hunk 5 above)
Comment (robbie-c) [reply frontend/src/queries/schema.ts:403]: I don't 100% agree, you might want to use this with a large container to show a ton of data, e.g. if you had 54 columns to show weekly retention over a year. We could do something smart like try to figure out whether what the user has asked for fits inside the container, and use the small layout if it doesn't, but I don't want to overengineer this.

File: frontend/src/scenes/web-analytics/webAnalytics.scss (hunk 4 above)
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalytics.scss]: Yep, done

File: frontend/src/queries/schema.ts (hunk 5 above)
Comment (benjackwhite) [reply frontend/src/queries/schema.ts:403]: Fair enough. Then lets go with it for now

**Review by benjackwhite — APPROVED** [review]

**Pull request merged**